src/Microsoft.OData.Core/Json/JsonReaderExtensions.cs

```diff
@@ -125,6 +126,16 @@ internal static string ReadStringValue(this IJsonReader jsonReader)
         }
 
         /// <summary>
+        /// Reads the next node from the <paramref name="jsonReader"/> as a URI and verifies that it is a PrimitiveValue node of type string.

```

Comment:
... and verifies that is is a **PrimitiveValue** (?)  node

Reply:
URIs are represented in JSON as strings.  ReadStringValue verifies that the JSON node is a primitive value node and is a string.

---
In reply to: [150044750](https://github.com/OData/odata.net/pull/989#discussion_r150044750) [](ancestors = 150044750)